hera_cal/lstbin.py

```diff
@@ -433,52 +433,50 @@ def config_lst_bin_files(data_files, filetype='uvh5', dlst=None, atol=1e-10, lst
     data_files : type=list of lists: nested set of lists, with each nested list containing
                  paths to data files from a particular night. These files should be sorted
                  by ascending Julian Date. Frequency axis of each file must be identical.
-    filetype : str, filetype of filepaths in data_files. options=['miriad', 'uvh5']
     dlst : type=float, LST bin width. If None, will get this from the first file in data_files.
     lst_start : type=float, starting LST for binner as it sweeps from lst_start to lst_start + 2pi.
     fixed_lst_start : type=bool, if True, LST grid starts at lst_start, regardless of LST of first data
         record. Otherwise, LST grid starts at LST of first data record.
     ntimes_per_file : type=int, number of LST bins in a single output file
 
-    Returns (lst_grid, dlst, file_lsts, start_lst)
+    Returns
     -------
     lst_grid : float ndarray holding LST bin centers
     dlst : float, LST bin width of output lst_grid
     file_lsts : list, contains the lst grid of each output file
     start_lst : float, starting lst for LST binner
+    lst_arrays : list, list of lst arrays for each file
+    time_arrays : list, list of time arrays for each file
     """
     # get dlst from first data file if None
     if dlst is None:
-        start, stop, int_time = io.get_file_lst_range(data_files[0][0], filetype=filetype)
-        dlst = int_time
-
-    # get start and stop times for each list of files in data_files.
-    # add_int_buffer adds an integration to the end time of df[:-1] files,
-    # and the %(2pi) ensures everything is within a 2pi LST grid.
-    data_times = []
-    for df in data_files:
-        data_times.append(np.array(io.get_file_lst_range(df, add_int_buffer=True, filetype=filetype))[:2, :].T % (2 * np.pi))
-
-    # unwrap data_times less than lst_start, get starting and ending lst
+        dlst, dtime, lst_arr, time_arr = io.get_file_times(data_files[0][0], filetype='uvh5')
+
+    # get time arrays for each file
+    lst_arrays = []
+    time_arrays = []
+    for dfs in data_files:
+        dlsts, dtimes, larrs, tarrs = io.get_file_times(dfs, filetype='uvh5')
+        lst_arrays.append(larrs)
+        time_arrays.append(tarrs)
+    lst_arrays = np.asarray(lst_arrays)
+    time_arrays = np.asarray(time_arrays)
+
+    # unwrap times relative to lst_start, and get global start and end lst
     start_lst = 100
```

Comment:
I find it very confusing to have two variables, one called `start_lst` and the other `lst_start`. Can we disambiguate them somehow?

Reply:
if using `fixed_lst_start` they are the same. one is just a dummy variable to keep track of the actual starting point of the files, the other is a preset for the starting output LST files.

Reply:
Can we add a _ to the front of the dummy variable?